src/storage/src/hummock/sstable_store.rs

```diff
@@ -310,6 +310,71 @@ impl SstableStore {
         )))
     }
 
+    pub fn support_prefetch(&self) -> bool {
+        self.large_query_memory_usage > 0
+    }
+
+    pub async fn get_with_prefetch(
+        &self,
+        sst: &Sstable,
+        block_index: usize,
+        max_prefetch_block_count: usize,
+        policy: CachePolicy,
```

Comment:
How the provided cache policy is used is confusing. The behavior is:
1. If no prefetch is triggered, the requested block is inserted into block cache with the provided `policy`
2. If prefetch is triggered, the requested block as well as the prefetched blocks are inserted into block cache with `CachePolicy::Fill(CachePriority::Low)`

How about making it more consistent?

Reply:
More thoughts:
we have two mechanism to constrain prefetch memory usage now:
- For prefetch with streaming read, `pending_streaming_loading` is used to track the usage and it is limited by `large_query_memory_usage`
- For prefetch with normal read, block cache is used to track the usage and it is limited by `block_cache_capacity` (and the low priority ratio).


The motivation of introducing low priority block cache is to control the behavior of cache refill, not block prefetch, and we are mixing it up here. IMO, logically it is better to have a clear sepration:
- Low priority block cache: enable block reuse *across queries* without polluting the whole block cache. Blocks in here can be evicted.
- Prefetch buffer: buffer prefetched blocks *within a query*. Blocks in here cannot be eviected.

Ideas:
1. rename `pending_streaming_loading` to `prefetch_buffer_usage` and `large_query_memory_usage` to `prefetch_buffer_capacity`
2. Prefetched blocks will be put into low priority block cache and the block holder will be dropped immediately
3. Prefetched blocks will be put into a buf (holding Bytes, not the BlockHolder) and account for `prefetch_buffer_usage`. For normal read prefetch, the buf is a `Vec<Block>`. For streaming read prefetch, the buf is the implicit recv buf.
4. When there is no quota in `prefetch_buffer_capacity`, prefetch will be disable and fall back to per block normal read.


What do you think?